### PR TITLE
My VA | Adds "pdf_support" flag for /v0/my_va/submission_statuses

### PR DESCRIPTION
**Pull request by DanielMiller-agile6**
## Summary

- *This work is behind a feature toggle (flipper)*: YES (`my_va_form_submission_pdf_link`)
- Adds a `pdf_support` flag to submission status metadata to let FE know if a PDF download is possible for a submitted form
- Reused `SubmissionPdfUrlService` (moved/renamed as `Forms::SubmissionStatuses::PdfUrls`) to determine if a form-id supports an archived PDF download in order to keep logic synced with `/v0/my_va/submission_pdf_urls` implementation
- Appends the new `pdf_support` flag in `SubmissionStatusesController` to avoid adding app-specific logic to `/lib/forms`
- Team: AuthExp - My VA

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/98369
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/81849

## Testing done

- [x] *New code is covered by unit tests*
- Old behavior did not include the `pdf_support` flag in the response data for `/v0/my_va/submission_statuses`. No other functionality is changed.
- Verified locally with console. Updated tests and verified passing
- existing specs test for Flipper behavior
- This is experimental functionality for My VA.  We're in the process of integrating this into our UI, and will test in a fully integrated environment once all pieces are on stage, behind feature flag.  Coordinating with VFF team when testing with real PDF files saved to S3.

## What areas of the site does it impact?
My VA

## Screenshot
Updated Swagger docs:
Screenshot 2024-12-12 at 2 08 43 PM

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- ~[ ]  Documentation has been updated (link to documentation)~
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- ~[ ]  Feature/bug has a monitor built into Datadog (if applicable)~ (monitors will be created after testing and refinement on staging)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- ~[ ]  I added a screenshot of the developed feature~

## Requested Feedback

Best practices & organization: Maybe there's a way to clean this up to move some logic out of the controller?

### CODE REVIEW

File: app/controllers/v0/my_va/submission_statuses_controller.rb [hunk 1]
```diff
@@ -24,6 +24,19 @@ def allowed_forms
         %w[20-10206 20-10207 21-0845 21-0972 21-10210 21-4142 21-4142a 21P-0847]
       end
 
+      def apply_pdf_support(result)
```
Comment (tpharrison) [inline app/controllers/v0/my_va/submission_statuses_controller.rb:27]: Can this method be moved to the report class and called form within it?

Comment (DanielMiller-agile6) [reply app/controllers/v0/my_va/submission_statuses_controller.rb:27]: i can try it out 👍 . I was unsure if there would be problems if we were to import functionality from `/app` into a module in '/lib'.  Would there be any concerns with that?  If so, I might also be able to move the `SubmissionPdfUrlService` into `/lib/forms/submission_status` as well.

Comment (tpharrison) [reply app/controllers/v0/my_va/submission_statuses_controller.rb:27]: I was thinking the `Report` class would depend on the `SubmissionPdfUrlService` class.  Let's hop on a call and hash it out.

File: lib/forms/submission_statuses/formatter.rb [hunk 2]
```diff
@@ -46,7 +48,9 @@ def build_submissions_map(submissions)
             message: nil,
             status: nil,
             created_at: submission.created_at,
-            updated_at: nil
+            updated_at: nil,
+            pdf_support: PdfUrls.new(form_id: submission.form_type,
```
Comment (tpharrison) [inline lib/forms/submission_statuses/formatter.rb:52]: Just a suggestion.  Update this...
```ruby
  pdf_support: PdfUrls.new(form_id: submission.form_type,
                           submission_guid: submission.benefits_intake_uuid).supported?
```

To someting like this...
```ruby
pdf_support: pdf_supported?(submission)
```
```ruby
def pdf_supported?(submission)
  PdfUrls.new(
    form_id: submission.form_type,
    submission_guid: submission.benefits_intake_uuid
  ).supported?
end
```

Comment (tpharrison) [reply lib/forms/submission_statuses/formatter.rb:52]: Also, I'm not sure the class name `Formatter` is the right choice (I originally named it), but we can rename in the future if necessary.

Comment (DanielMiller-agile6) [reply lib/forms/submission_statuses/formatter.rb]: agree... can't think of a better name atm 🤔 .  It's more like a "Submission Status Collection Builder"? I'll leave as-is for now 👍

**Review by ericboehs — APPROVED** [review]

**Pull request merged**